apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.js

```diff
@@ -65,41 +94,52 @@ const GetAllResidentBillingReceiptsService = new GQLCustomSchema('GetAllResident
-                const billingReceipts = []
-                for (let i = 0; i < allServiceConsumers.length; ++i) {
-                    const receiptsQuery = { ...receiptsWhere, ...{ account: { id: allServiceConsumers[i].billingAccount.id } } }
 
-                    const billingReceiptsForConsumer = await BillingReceipt.getAll(
+                const processedReceipts = []
+                for (const serviceConsumer of serviceConsumers) {
+
+                    const receiptsQuery = { ...receiptsWhere, ...{ account: { id: serviceConsumer.billingAccount.id } }, deletedAt: null }
+                    const receiptsForConsumer = await BillingReceipt.getAll(
                         context,
                         receiptsQuery,
                         {
                             sortBy, first, skip,
                         }
                     )
 
-                    billingReceipts.push(
-                        billingReceiptsForConsumer.map(
-                            receipt => ({
-                                id: receipt.id,
-                                dv: receipt.dv,
-                                recipient: receipt.recipient,
-                                period: receipt.period,
-                                toPay: receipt.toPay,
-                                toPayDetails: receipt.toPayDetails,
-                                services: receipt.services,
-                                printableNumber: receipt.printableNumber,
-                                serviceConsumer: allServiceConsumers[i],
-                                currencyCode: get(receipt, ['context', 'integration', 'currencyCode'], null),
-                            })
-                        ))
+                    receiptsForConsumer.forEach(receipt => processedReceipts.push({
+                        id: receipt.id,
+                        dv: receipt.dv,
+                        recipient: receipt.recipient,
+                        period: receipt.period,
+                        toPay: receipt.toPay,
+                        toPayDetails: receipt.toPayDetails,
+                        services: receipt.services,
+                        printableNumber: receipt.printableNumber,
+                        serviceConsumer: serviceConsumer,
+                        currencyCode: get(receipt, ['context', 'integration', 'currencyCode'], null),
+                    }))
+                }
+
+
+                const receiptsWithPayments = []
+                for (const processedReceipt of processedReceipts) {
+
+                    const organizationId = get(processedReceipt.serviceConsumer, ['organization', 'id'])
+                    const accountNumber = get(processedReceipt.serviceConsumer, ['accountNumber'])
+
+                    receiptsWithPayments.push(({
+                        ...processedReceipt,
+                        paid: await getPaymentsSum(context, organizationId, accountNumber, processedReceipt.period),
```

Comment:
Ok for now....But it's 1-query per receipt
Can be optimised if doing something like `getAll` with like `receipt: {id_in: [....]}` and then parsing it

Reply:
This will decrease amount of queries, but if you decided not to do it since its priority task...at least add TODO for next week or so :)

Reply:
Yes, this would be nice

Reply:
https://doma.atlassian.net/browse/DOMA-1793